### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by gunbux**
Resolves: #239, #243

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -244,6 +255,10 @@ instead opting to defer the creation of the new `Task` to the `Model` itself. Th
 code to parse in all the fields of `Task` and leaving student as a list of Strings, which we then use to search through
 the `Model` to find the students that are assigned to the task.
 
+Show below is an activity diagram of how a task is edited.
+
+![Task Edit Activity Diagram](images/TaskEditActivityDiagram.png)
```
Comment (tensaida) [inline docs/DeveloperGuide.md:273]: There is no `TaskEditActivityDiagram.png`. You may have forgotten to add the file.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -677,29 +654,50 @@ testers are expected to do more *exploratory* testing.
...
+
+#### Tutorial group management
+1. Creating and managing tutorial groups
+   1. Create a new tutorial group with the format TXX. `tutorial add g/T01` 
+   2. See a list of tutorial groups using `tutorial list` <br>
+      Expected: A list of tutorial groups is shown.
+   3. Add a student to the tutorial group with the format `student enroll 1 g/T01`
+   4. The UI should be updated with the new student list with update tutorial group info.
+   5. Filter the student list by tutorial group with the format `tutorial filter g/T01`
+   6. Clear the filter with the format `student unfilter`
```
Comment (tensaida) [inline docs/DeveloperGuide.md:705]: Need to add grades

**Review by tensaida — CHANGES_REQUESTED** [review]

**Comment by tensaida** [comment]
This commit does not resolve #239

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (gunbux) [reply docs/DeveloperGuide.md:705]: Added grade in the earlier manual testing case

**Comment by gunbux** [comment]
> This commit does not resolve #239

indeed

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -693,36 +670,83 @@ testers are expected to do more *exploratory* testing.
...
+#### Task management
+1. Creating, removing, grading and viewing tasks
+
+    1. Create a student with the name "Assignment 1" using `task add tn/Assignement 1 i/Complete Assignment 1 d/10/10/2022`. <br>
+    2. Click on the `Tasks` Button on the left side of the screen. <br>
+       Expected: The UI is updated with the new task list.
+    3. Edit the student with the command `task edit 1 s/David Li`
+    4. Double click on the task to view the task details.
+       Expected: The details shows the list of students assigned to the task.
+    5. Grade the student with the command `grade edit 1 1 gr/T`
```
Comment (tensaida) [inline docs/DeveloperGuide.md:692]: I see it 👁

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -693,36 +670,83 @@ testers are expected to do more *exploratory* testing.
...
+
+#### Help
+1. Type `help me` for a link to the user guide. <br>
+   Expected: A new window is opened with the user guide link.
+
+#### Exiting the program
+1. Type `bye bye` <br>
+   Expected: The application exits. Data is automatically saved.
+
+## Appendix: Effort
```
Comment (tensaida) [inline docs/DeveloperGuide.md:717]: Good job!

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -693,36 +670,83 @@ testers are expected to do more *exploratory* testing.
...
+Justification for the difficulty level: 
+1. Moving parts - While AB3 uses only one entity type, Persons, which was adapted to be our Student, we have
+multiple entities, namely Student, Task, Tutorial Group, and Grade. This means that we have to implement
+multiple classes, and multiple classes that interact with each other.
+2. Different purposes - While _some_ of the code for AB3 commands could be reused, there were many cases where
+the code had to be redesigned and rewritten in order to fit the purpose of our application.
+    1. For example, the Task Edit Command had to be completely rewritten because we required the ability to 
+search through the current model's list of students using a list of strings, and then generating a CommandOutput
+from that
+    2. Other examples include the all the Grade Commands, which had to be written from scratch, as well as the
```
Comment (tensaida) [inline docs/DeveloperGuide.md:739]: Used a map, a data structure not used in the AB3

**Review by tensaida — CHANGES_REQUESTED** [review]

**Review by tensaida — APPROVED** [review]
LGTM!

**Pull request merged**